test/common/stats/stats_impl_test.cc

```diff
@@ -170,7 +158,32 @@ class DefaultTagRegexTester {
                        stat_name);
   }
 
-  std::vector<TagExtractorPtr> tag_extractors_;
+  /**
+   * Reimplements TagProducerImpl::produceTags, but extracts the tags in reverse order.
+   * His helps demonstrate that the order extractors does not matter to the end result,
+   * assuming we don't care about tag-order. This is in large part correct by design because
+   * stat_name is not mutated until all the extraction is done.
+   * @param metric_name std::string a name of Stats::Metric (Counter, Gauge, Histogram).
```

Comment:
There's no parameter called `metric_name`, did you mean stat_name?

Reply:
I renamed the var to metric_name for consistency with the interface in Stats.